offline/cable_cru_TRENDY.F90

```diff
@@ -1,1318 +1,1101 @@
+  ErrStatus = nf90_get_var(FileID, LatID, Latitudes)
+
+  ALLOCATE(Longitudes(xDimSize))
+  ErrStatus = nf90_inq_varid(FileID, 'longitude', LonID)
+  ErrStatus = nf90_get_var(FileID, LonID, Longitudes)
+
+  ! Now read the mask, this mask is an integer (0/1) version
+  ALLOCATE(mask(xDimSize, yDimSize))
+  ALLOCATE(LandMask(xDimSize, yDimSize))
+  ALLOCATE(CRU%LandMask(xDimSize, yDimSize))
+  ErrStatus = nf90_inq_varid(FileID, 'land', LandID)
+  ErrStatus = nf90_get_var(FileID, LandID, LandMask)
+
+  ! Finally close the mask file
+  ErrStatus = nf90_close(FileID)
+
+  ! For some reason we also want a logical version
+  !CRU%LandMask = to_logical(mask)
+  WHERE (LandMask > 0)
+    CRU%LandMask = .TRUE.
+    mask = 1
+  ELSEWHERE
+    CRU%LandMask = .FALSE.
+    mask = 0
+  END WHERE
+
+  ! Set the total number of land cells by counting the TRUEs in the landmask.
+  CRU%mLand = COUNT(CRU%LandMask)
+
+  ! Now we know how many elements in our specific region, allocate global (!!!)
+  ! arrays for latitude and longitude value/indices.
+  ALLOCATE(Latitude(CRU%mLand), Longitude(CRU%mLand))
+  ALLOCATE(Land_y(CRU%mLand), Land_x(CRU%mLand))
+
+  ! Now pull out the coordinates in our current mask
+  MaskCounter = 1
+  LoopThroughLats: DO LatIndx = 1, yDimSize
+    LoopThroughLons: DO LonIndx = 1, xDimSize
+      IF (CRU%LandMask(LonIndx, LatIndx)) THEN
+        Land_x(MaskCounter) = LonIndx
+        Land_y(MaskCounter) = LatIndx
+        Longitude(MaskCounter) = Longitudes(LonIndx)
+        Latitude(MaskCounter) = Latitudes(LatIndx)
+        MaskCounter = MaskCounter + 1
+      END IF
+    END DO LoopThroughLons
+  END DO LoopThroughLats
+
+  ! Allocate memory for the actual meteorological forcing data
+  AllocateMemory: DO VarIndx = 1, SIZE(CRU%Met)
+    ALLOCATE(CRU%Met(VarIndx)%MetVals(CRU%mLand))
+  END DO AllocateMemory
+
+  ! And do the same for nitrogen deposition and atmospheric CO2
+  ALLOCATE(CRU%NDepVals(CRU%mLand))
+  ALLOCATE(CRU%CO2Vals(CRU%mLand))
+
+  ! Allocate global data- this practice must be purged in future.
+  metGrid = "mask"
+  mLand = CRU%mLand
+  nMetPatches = 1
+
+  ALLOCATE(Lat_all(xDimSize, yDimSize), Lon_all(xDimSize, yDimSize))
+  FillLatitudes: DO LonIndx = 1, xDimSize
+    Lat_all(LonIndx, :) = Latitudes
+  END DO FillLatitudes
+
+  FillLongitudes: DO LatIndx = 1, yDimSize
+    Lon_all(:, LatIndx) = Longitudes
+  END DO FillLongitudes
+
+  ! Some CABLE time units?
+  shod = 0.
+  sdoy = 1
+  smoy = 1
+  syear = CRU%CYEAR
+
+  ! Allocate memory for the average long-wave radiation down for some reason?
+  ALLOCATE(CRU%AVG_LWDN(mLand))
+END SUBROUTINE read_landmask
+
+!!------------------------------------------------------------------------------!
+
+SUBROUTINE prepare_temporal_dataset(FileName, TargetArray)
+  !*## Purpose
+  !
+  ! Read the CO2 dataset.
+  !
+  !## Method
+  !
+  ! Read the ```FileName``` text file which contains the yearly CO2 data in
+  ! ```YEAR VALUE``` pairs into ```TargetArray```.
+
```

Comment:
If the name is generic, so should the description. Nothing in the name of the subroutine refers to CO2 specifically, so let's remove it from the description.


```suggestion
  ! Read a dataset varying in time but constant in space.
  !
  !## Method
  !
  ! Read the ```FileName``` text file which contains the time-varying data in
  ! ```TIME VALUE``` pairs into ```TargetArray```.

```